amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/fileplans/FilePlanTests.java

```diff
@@ -594,4 +603,178 @@ public void listHolds()
+        assertEquals(roleModelList.size(), 5);
+    }
+
+    /**
+     * <pre>
+     * Given that a file plan exists
+     * When a non-RM user asks the API for the roles and relation
+     * Then the status code 403 (Permission denied) is return
+     * </pre>
+     */
+    @Test
+    public void nonRmUserFilePlanRolesAndRelation()
+    {
+        // Create a random user
+        UserModel nonRMuser = getDataUser().createRandomTestUser("testUser");
+        // Call to new API to get the roles and capabilities
+        getRestAPIFactory().getFilePlansAPI(nonRMuser).getFilePlanRolesAndRelation(FILE_PLAN_ALIAS, "");
+        assertStatusCode(FORBIDDEN);
+    }
+
+    /**
+     * <pre>
+     * Given that a file plan exists
+     * When a RM_Manager user asks the API for the roles and relation
+     * returns the RM_Manager role and capabilities
+     * </pre>
+     */
+    @Test
+    public void rmManagerFilePlanRolesAndCapabilities()
+    {
+        // Create a random user
+        UserModel managerUser = getDataUser().createRandomTestUser("managerUser");
+        // Assign RecordsManager role to user
+        getRestAPIFactory().getRMUserAPI().assignRoleToUser(managerUser.getUsername(), ROLE_RM_MANAGER.roleId);
+        String parameters = "where=(personId='" + managerUser.getUsername() + "')";
+        // Call to new API to get the roles and capabilities
+        Object response = getRestAPIFactory().getFilePlansAPI(managerUser).getFilePlanRolesAndRelation(FILE_PLAN_ALIAS, parameters);
+        assertStatusCode(OK);
+        Map<?, ?> roleModelMap = getRoleModelMap(response);
+        assertEquals((String) roleModelMap.get("displayLabel"), ROLE_RM_MANAGER.displayName);
+    }
+
+    /**
+     * <pre>
+     * Given that a file plan exists
+     * When a RM_User asks the API for the roles and relation
+     * returns the RM_User role and capabilities
+     * </pre>
+     */
+    @Test
+    public void rmUserFilePlanRolesAndCapabilities()
+    {
+        // Create a random user
+        UserModel rmUser = getDataUser().createRandomTestUser("rmUser");
+        // Assign rmUser role to user
+        getRestAPIFactory().getRMUserAPI().assignRoleToUser(rmUser.getUsername(), ROLE_RM_USER.roleId);
+        String parameters = "where=(personId='" + rmUser.getUsername() + "')";
+        // Call to new API to get the roles and capabilities
+        Object response = getRestAPIFactory().getFilePlansAPI(rmUser).getFilePlanRolesAndRelation(FILE_PLAN_ALIAS, parameters);
+        assertStatusCode(OK);
+        Map<?, ?> roleModelMap = getRoleModelMap(response);
+        assertEquals((String) roleModelMap.get("displayLabel"), ROLE_RM_USER.displayName);
+    }
+
+    /**
+     * <pre>
+     * Given that a file plan exists
+     * When a RM_Power_User asks the API for the roles and relation
+     * returns the RM_User role and capabilities
+     * </pre>
+     */
+    @Test
+    public void rmPowerUserFilePlanRolesAndCapabilities()
```

Comment:
I think `rmPowerUserFilePlanRolesAndCapabilities` and `rmUserFilePlanRolesAndCapabilities` check the same scenario as `rmManagerFilePlanRolesAndCapabilities`. Alternatively, we can remove that 2 tests and create one test to assign two roles to a user and check if the endpoint returns exactly 2 roles assigned.